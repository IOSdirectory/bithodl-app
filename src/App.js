import React, { Component } from 'react';
import { StatusBar } from 'react-native';
import PropTypes from 'prop-types';
import { Provider, connect } from 'react-redux';
import { addNavigationHelpers } from 'react-navigation';

import AppNavigator from './AppNavigator';
import getStore from './getStore';

const navReducer = (state, action) => {
  const newState = AppNavigator.router.getStateForAction(action, state);
  return newState || state;
};

const store = getStore(navReducer);

@connect((state) => ({
  nav: state.nav
}))
class AppWithNavigationState extends Component {
  render() {
    return (
      <AppNavigator
        navigation={addNavigationHelpers({
          dispatch: this.props.dispatch,
          state: this.props.nav
        })}
      />
    );
  }
}

AppWithNavigationState.propTypes = {
  dispatch: PropTypes.func,
  nav: PropTypes.object
};

export default class App extends Component {
  constructor() {
    super(...arguments);

    StatusBar.setHidden(false);
    StatusBar.setBarStyle('default');
  }

  render() {
    return (
      <Provider store={store}>
        <AppWithNavigationState />
      </Provider>
    );
  }
}
