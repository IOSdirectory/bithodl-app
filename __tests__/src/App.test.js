import React from 'react';
import { StatusBar } from 'react-native';
import renderer from 'react-test-renderer';
import App from '../../src/App';

jest.mock('StatusBar', () => {
  return {
    setHidden: jest.fn(),
    setBarStyle: jest.fn()
  };
});

describe('App', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <App />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('contructor', () => {
    beforeEach(() => {
      renderer.create(<App />);
    });

    it('shows the status bar', () => {
      expect(StatusBar.setHidden).toHaveBeenCalledWith(false);
    });

    it('sets the status bar style to "default"', () => {
      expect(StatusBar.setBarStyle).toHaveBeenCalledWith('default');
    });
  });
});
