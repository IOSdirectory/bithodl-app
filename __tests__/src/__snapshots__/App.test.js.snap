// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column-reverse",
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#E9E9EF",
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 5,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column-reverse",
                },
                Object {
                  "backgroundColor": "#000",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={undefined}
                pointerEvents="auto"
                style={
                  Object {
                    "backgroundColor": "#E9E9EF",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "shadowColor": "black",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0.2,
                    "shadowRadius": 5,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#ffffff",
                            "flex": 1,
                            "justifyContent": "center",
                            "padding": 40,
                          },
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fec98a",
                            "flex": 1,
                            "justifyContent": "center",
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "height": 42,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "zIndex": 1000,
                          }
                        }
                      />
                      <Image
                        source={1}
                        style={
                          Object {
                            "height": 95,
                            "width": 64,
                          }
                        }
                      />
                      <View
                        pointerEvents={undefined}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#ffffff",
                              "bottom": 30,
                              "left": 40,
                              "position": "absolute",
                              "right": 40,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ]
                        }
                      >
                        <ActivityIndicator
                          animating={true}
                          color="#D59247"
                          hidesWhenStopped={true}
                          size="small"
                          style={
                            Object {
                              "height": 42,
                            }
                          }
                        />
                      </View>
                      <View
                        pointerEvents="none"
                        style={
                          Object {
                            "alignItems": "center",
                            "bottom": 0,
                            "flex": 1,
                            "justifyContent": "center",
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "zIndex": 2000,
                          }
                        }
                      >
                        <VibrancyView
                          blurAmount={5}
                          blurType="dark"
                          pointerEvents="none"
                          style={
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                              },
                              Array [
                                Object {
                                  "bottom": 0,
                                  "left": 0,
                                  "opacity": 1,
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                },
                                Object {
                                  "opacity": 0,
                                },
                              ],
                            ]
                          }
                        />
                        <View
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#ffffff",
                                "borderRadius": 12,
                                "marginTop": 0,
                                "shadowColor": "#000000",
                                "shadowOffset": Object {
                                  "height": 0,
                                  "width": 0,
                                },
                                "shadowOpacity": 0.3,
                                "shadowRadius": 12,
                                "width": 710,
                              },
                              Object {
                                "marginTop": -500,
                                "opacity": 0,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "backgroundColor": "#FFD6D6",
                                "borderTopLeftRadius": 12,
                                "borderTopRightRadius": 12,
                                "padding": 15,
                                "paddingTop": 17,
                              }
                            }
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              disabled={false}
                              ellipsizeMode="tail"
                              numberOfLines={undefined}
                              style={
                                Array [
                                  Object {
                                    "color": "#504F63",
                                    "fontFamily": "Avenir Next",
                                    "fontWeight": "400",
                                  },
                                  Object {
                                    "color": "#FF5A59",
                                    "fontSize": 16,
                                    "fontWeight": "600",
                                    "textAlign": "center",
                                  },
                                ]
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={false}
                                disabled={false}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "color": undefined,
                                      "fontSize": 12,
                                    },
                                    Object {
                                      "fontSize": 28,
                                    },
                                    Object {
                                      "fontFamily": "Ionicons",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                  ]
                                }
                              >
                                
                              </Text>
                            </Text>
                          </View>
                          <View
                            style={
                              Object {
                                "padding": 25,
                                "paddingLeft": 30,
                                "paddingRight": 30,
                              }
                            }
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              disabled={false}
                              ellipsizeMode="tail"
                              numberOfLines={undefined}
                              style={
                                Array [
                                  Object {
                                    "color": "#504F63",
                                    "fontFamily": "Avenir Next",
                                    "fontWeight": "400",
                                  },
                                  Array [
                                    Object {
                                      "color": "#8F8E94",
                                      "fontSize": 16,
                                      "marginBottom": 20,
                                      "textAlign": "center",
                                    },
                                    Object {
                                      "marginBottom": 0,
                                    },
                                  ],
                                ]
                              }
                            >
                              
                            </Text>
                          </View>
                          <View>
                            <View
                              accessibilityComponentType={undefined}
                              accessibilityLabel={undefined}
                              accessibilityTraits={undefined}
                              accessible={true}
                              hasTVPreferredFocus={undefined}
                              hitSlop={undefined}
                              isTVSelectable={true}
                              nativeID={undefined}
                              onLayout={undefined}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                  },
                                  Object {
                                    "alignItems": "center",
                                    "borderBottomLeftRadius": 12,
                                    "borderBottomRightRadius": 12,
                                    "borderColor": "#fbfbfb",
                                    "borderTopWidth": 1,
                                    "padding": 15,
                                  },
                                ]
                              }
                              testID={undefined}
                              tvParallaxProperties={undefined}
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                disabled={false}
                                ellipsizeMode="tail"
                              >
                                <Text
                                  accessible={true}
                                  allowFontScaling={true}
                                  disabled={false}
                                  ellipsizeMode="tail"
                                  numberOfLines={undefined}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#504F63",
                                        "fontFamily": "Avenir Next",
                                        "fontWeight": "400",
                                      },
                                      Object {
                                        "color": "#0073FF",
                                        "fontWeight": "600",
                                        "letterSpacing": 1,
                                      },
                                    ]
                                  }
                                >
                                  OK
                                </Text>
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
